ldi-core/ldes-client/tree-node-supplier/src/main/java/ldes/client/treenodesupplier/repository/inmemory/InMemoryTreeNodeRecordRepository.java

```diff
@@ -54,12 +67,15 @@ public void destroyState() {
 		immutable = new HashSet<>();
 	}
 
-	public Optional<TreeNodeRecord> getOneTreeNodeRecordWithStatus(TreeNodeStatus treeNodeStatus) {
+	public Optional<TreeNodeRecord> getTreeNodeRecordWithStatusAndEarliestNextVisit(TreeNodeStatus treeNodeStatus) {
 		return switch (treeNodeStatus) {
 			case NOT_VISITED -> notVisited.isEmpty() ? Optional.empty() : Optional.of(notVisited.get(0));
 			case MUTABLE_AND_ACTIVE ->
 				mutableAndActive.isEmpty() ? Optional.empty() : Optional.of(mutableAndActive.poll());
-			case IMMUTABLE -> immutable.isEmpty() ? Optional.empty() : immutable.stream().findFirst();
+			case IMMUTABLE_WITH_UNPROCESSED_MEMBERS
+					-> almostImmutable.isEmpty() ? Optional.empty() : almostImmutable.stream().findFirst();
+			case IMMUTABLE_WITHOUT_UNPROCESSED_MEMBERS
+					-> immutable.isEmpty() ? Optional.empty() : immutable.stream().findFirst();
```

Comment:
Previous code = first "->" , then newline. New code has new layout?

Reply:
new code was too long to be readable on one line without a widescreen